lib/services/image-service.ts

```diff
@@ -0,0 +1,167 @@
+				this.pushImageToTable(table, imageDefinition.Platform, firstIcon, firstSplash);
+			}
+
+			for (let i = 1; i < maxLength; ++i) {
+				this.pushImageToTable(table, '', imageDefinition.Icons[i], imageDefinition.SplashScreens[i]);
+			}
+			
+			table.push(['', '', '']);
+		});
+		
+		this.$logger.out(table.toString());
+	}
+	
+	public promptForImageInformation(force: boolean): IFuture<void> {
+		let imagePath = this.$prompter.getString('Enter image file path:').wait(),
+			imageOptions = ['Icons', 'SplashScreens'],
+			chosenOption = this.$prompter.promptForChoice(`What type of resources do you want to create?`, imageOptions).wait(),
+			imageType = Server.ImageType.Icon;
+
+		if (chosenOption === imageOptions[1]) {
+			imageType = Server.ImageType.SplashScreen;
+		}
+		
+		return this.generateImages(imagePath, imageType, force);
+	}
+
+	public generateImages(initialImagePath: string, imageType: Server.ImageType, force: boolean): IFuture<void> {
+		return (() => {
+			this.validateImage(initialImagePath).wait();
+
+			temp.track();
+			let inputImageStream = this.$fs.createReadStream(initialImagePath),
+				tempDir = temp.mkdirSync('ab-images-'),
+				resultImageArchivePath = path.join(tempDir, 'images.zip'),
+				resultImageArchiveStream = this.$fs.createWriteStream(resultImageArchivePath);
+
+			this.replaceAll = force;
+			this.$logger.printInfoMessageOnSameLine('Generating images');
+			this.$progressIndicator.showProgressIndicator(this.$server.images.generateArchive(imageType, inputImageStream, resultImageArchiveStream), 2000).wait();
+			this.$logger.printInfoMessageOnSameLine('Extracting images');
+			this.$progressIndicator.showProgressIndicator(this.$fs.unzip(resultImageArchivePath, tempDir), 2000).wait();
+			this.$fs.unzip(resultImageArchivePath, tempDir).wait();
+
+			let imageBasePath = path.join(tempDir, 'App_Resources'),
+				images = this.$fs.enumerateFilesInDirectorySync(imageBasePath);
+
+			_.each(images, imagePath => {
+				if (this.$project.projectData.Framework === this.$projectConstants.TARGET_FRAMEWORK_IDENTIFIERS.NativeScript &&
+					~imagePath.indexOf(this.$devicePlatformsConstants.WP8)) {
+					// NativeScript does not support WP8 yet
+					return;
+				}
+
+				let projectImagePath = path.join(this.$project.appResourcesPath().wait(), imagePath.substring(imageBasePath.length));
+				this.copyImageToProject(imagePath, projectImagePath).wait();
+			});
+		}).future<void>()();
+	}
+
+	private copyImageToProject(imagePath: string, projectImagePath: string): IFuture<void> {
+		return (() => {
+			this.$fs.ensureDirectoryExists(path.dirname(projectImagePath)).wait();
+
+			if (this.replaceAll || !this.$fs.exists(projectImagePath).wait()) {
+				return this.$fs.copyFile(imagePath, projectImagePath).wait();
+			}
+
+			let replaceOptions = ['Yes for all', 'Yes', 'No', 'No for all'],
+				chosenOption = this.$prompter.promptForChoice(`${projectImagePath} already exists, do you want to replace it?`, replaceOptions).wait();
```

Comment:
After some discussion it was decided that failing within the prompter should be introduced in a subsequent PR